### DISCUSSION TITLE
Feature request: custom starter-workflows for an org

### DISCUSSION THREAD

**Discussion by j3parker** [community/community#25764] [Actions]
<p>It would be really cool if we could add custom starter workflows for our org that would appear when using the “new workflow” UI for a repo in the same org.</p>
<p>Ideally we could have a set of private templates for internal projects.</p>
<p>Template repositories would sort of work but are a heavy-handed solution and can’t be used easily after the fact. It would be really nice if these showed up in the “new workflow” UI with the ones from starter-workflows.</p>

**Comment by chrispat** [comment]
<p>This and many other org level workflow features are absolutly on our roadmap.  At the moment our main focus in making sure Actions works with GitHub Enterprise Server.  Once that work is done you will start to see new features rollout.</p>

**Comment by MChevy422** [comment]
<p>Hello <span><a class="mention" href="/j3parker">@j3parker</a></span>, <br><br>Thank you for your feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.<br><br>Once again, thank you for your input!<br><br>Greatly appreciated,<br><br>MChevy422</p>

**Comment by j3parker** [comment]
<p>Thank you for all your hard work! Actions has been great!</p>

**Comment by j3parker** [comment, marked as answer]
<p>This was launched as org workflow templates: <a href="https://help.github.com/en/actions/configuring-and-managing-workflows/sharing-workflow-templates-within-your-organization" rel="nofollow noopener">https://help.github.com/en/actions/configuring-and-managing-workflows/sharing-workflow-templates-within-your-organization</a></p>
<p>It has one pretty surprising limitation:</p>
<blockquote>
<p>to use templates to create workflows in private repositories, the organization must be part of an enterprise or GitHub One plan.</p>
</blockquote>
<p>but I suppose that’s just how it’s going to be! :frowning:</p>

**Reply by dataplor-mike** [reply]
for anyone coming across this in 2023, [the enterprise docs](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/creating-starter-workflows-for-your-organization#creating-a-starter-workflow) also specify a public repo:
> If it doesn't already exist, create a new public repository named .github in your organization.

**Comment by tboulis** [comment]
<p>Can we use starter workflows if the <code>.github</code> repo is private?</p>

**Comment by jazanne** [comment]
I have follow the instructions described in the docs to set up start workflow templates for my organization (yes it is an enterprise account), but the workflows do not show up when I go to create an action in a new repo. Any thoughts on what I could be missing?

**Reply by lukewhiting** [reply]
So far I have found starter workflows are only usable inside an org if your orgs `.github` repo is set to public... This is less than idea for a lot of companies. Would love to find out if there's a way to host starter workflow templates in a private or internal repo.

**Comment by nyellavula-exelixis** [comment]
We are running into same issue, where we can't get starter workflows working due to `public` repository limitation. GitHub, Can't we have an `internal` .github repo ?

**Comment by nebuk89** [comment]
This is on our list! I know it's a way out, but we should have this some time early next year (just with where this sits behind other things that we are doing). But getting this done is something we hear a lot and will get to soon ❤️

**Reply by laughedelic** [reply]
Now that the roadmap ticket has been [deprecated](https://github.com/github/roadmap/issues/772), is this still on the backlog? 

The limitation of having starter workflows only as public is a blocker for many organizations.
